client/src/charts/wrapped_nivo/nivo_bubble.js

```diff
@@ -0,0 +1,242 @@
+    // note these aren't values for the specific node, but for the whole graph, e.g. r is always the outer circle radius,
+    // and (x, y) is the center of the graph. Node specific values are calculated below
+    r: graph_radius,
+    x: center_x,
+    y: center_y,
+
+    fill,
+    color,
+    borderColor,
+    borderWidth,
+  } = style;
+
+  const {
+    node_radius,
+    node_x,
+    node_y,
+  } = (() => {
+    if ( _.isNull(node.parent) ){
+      return {
+        node_radius: graph_radius,
+        node_x: center_x,
+        node_y: center_y,
+      };
+    } else {
+
+      // need to be clear here, node.value !== graph_data.value as seen below. The config data from graph_data is stored in
+      // node.data. The value in node.value is node.data.value PLUS the sum of child values
+      // ... kind of makes sense for the standard use of this graph, but still a bit of annoying hidden logic. Makes what
+      // we're doing here extra hacky
+      const proportion_ratio = node.value/node.parent.value;
+
+      const node_radius = _.max([
+        graph_radius*Math.sqrt(proportion_ratio), // do the math, make the actual area proportional
+        MIN_NODE_RADIUS,
+      ]);
+
+      // this y position will place the bottom of the inner circle just barely above the bottom of the outer circle.
+      // Easier to judge proptions than when it's centered, and the slight offset stops the the svg's edges from
+      // overlaping and looking jagged
+      const node_y = center_y + (graph_radius - node_radius) - 1;
+
+      return {
+        node_radius,
+        node_x: center_x,
+        node_y,
+      };
+    }
+  })();
+  
+  return (
+    <g transform={`translate(${node_x},${node_y})`}>
+      <circle
+        r={node_radius}
+        {...handlers}
+        fill={fill ? fill : color}
+        stroke={borderColor}
+        strokeWidth={borderWidth}
+        shapeRendering={"geometricPrecision"}
+      />
+    </g>
+  );
+};
+
+
+export class CircleProportionChart extends React.Component{
+  render(){
+    const {
+      margin,
+      is_money,
+      text_formatter,
+      labelSkipWidth,
+      height,
+      child_value,
+      child_name,
+      parent_value,
+      parent_name,
+      disable_table_view,
+    } = this.props;
+
+    const color_scale = d3.scaleOrdinal().range(newIBCategoryColors);
+    const value_formatter= get_formatter(is_money, text_formatter, true, true);
+
+    const graph_data = {
+      id: parent_name,
+      name: parent_name,
+      // ... nivo bubble will roll child values back up and add them to the parent value for use in the graph
+      // so we need to remove the inner portion from the total here
+      value: parent_value - child_value,
+      color: color_scale(parent_name),
+      children: [
+        {
+          id: child_name,
+          name: child_name,
+          value: child_value,
+          color: color_scale(child_name),
+        },
+      ],
+    };
+
+    const tooltip = () => (
```

Comment:
Is it possible to get `TooltipFactory` or `DefaultTooltip` from `nivo_common` and make use of those here in this case?